[Bug] test_llm_timeout.py CI 실패 - AsyncLLMWrapper None 체크 오류

## 문제

main 브랜치 CD 파이프라인(`Run tests (main - full)`)에서 1개 테스트 실패:

```
FAILED backend/scripts/testing/reliability/test_llm_timeout.py::TestSupervisorLLMTimeout::test_supervisor_llm_none_uses_rule_based
assert <AsyncLLMWrapper object> is None
```

`supervisor_node.llm`이 `None`이 아닌 `AsyncLLMWrapper` 객체로 감싸져 있어 테스트 assertion 실패.

## 원인

테스트가 `supervisor_node.llm is None`을 기대하지만, 실제 `SupervisorNode`는 LLM이 없어도 `AsyncLLMWrapper`로 감싸서 `.llm`이 절대 `None`이 아님.

## 해결 방법

테스트의 assertion을 실제 구현에 맞게 수정.

Ref: https://github.com/SKN19-FINAL-5team/LLM/actions/runs/21794328810

--- a/backend/scripts/testing/reliability/test_llm_timeout.py
+++ b/backend/scripts/testing/reliability/test_llm_timeout.py
@@ -115,12 +115,11 @@ class TestSupervisorLLMTimeout:
     """Supervisor LLM timeout should fallback to rule-based routing."""
 
     @pytest.mark.asyncio
-    async def test_supervisor_llm_none_uses_rule_based(self):
-        """SupervisorNode with llm=None uses rule-based routing."""
+    async def test_supervisor_llm_failure_uses_rule_based(self):
+        """SupervisorNode falls back to rule-based when LLM call fails."""
         from app.supervisor.nodes.supervisor import SupervisorNode
 
         supervisor = SupervisorNode(llm=None)
-        assert supervisor.llm is None
 
         state = {
             "user_query": "환불 가능한가요?",
@@ -133,8 +132,13 @@ class TestSupervisorLLMTimeout:
             "review": None,
         }
 
-        node_fn = supervisor.as_node()
-        result = await node_fn(state)
+        # LLM 호출을 실패시켜 rule-based fallback 검증
+        with (
+            patch.object(supervisor, "_primary_llm", None),
+            patch.object(supervisor, "_fallback_llm", None),
+        ):
+            node_fn = supervisor.as_node()
+            result = await node_fn(state)
 
         assert "supervisor" in result
         supervisor_state = result["supervisor"]

